test/sharness/lib/0001-Detect-type-of-aliasing-unsupported-by-some-shells.patch

```diff
@@ -0,0 +1,46 @@
+From 6a43df1919da862f56c9af867a28bf2d7858e5ca Mon Sep 17 00:00:00 2001
```

Comment:
I agree that the behavior of sharness is surprising, but I'm concerned that this will cause more problems than it fixes. It already has a bunch of special cases to avoid rejecting valid code.

I'd rather document this in the sharness README and move on unless we can find a more reliable way to do this.

Reply:
Nobody will ever discover the sharness README :)
No worries though - I will pull this commit out, and use it locally instead.
Unless you are open to have it gated on an environment variable...?

Reply:
Force-pushed a removal ( it was the first commit )

Reply:
Ideally, we'd trigger it when we get a failure. However, in this case, that wouldn't have helped regardless because we weren't checking for errors.

You're right that nobody will read the README... Honestly, it's _not_ terrible given that we can catch and deal with any false positives. What if we include it but put in a comment explaining exactly what it does, where it comes from (it's non-obvious that we're patching it in), and that it's a heuristic and that we should remove it if it causes problems?

Reply:
I'll open a separate PR once I had a chance to write up better prose. The the error will link to that.

Reply:
For now let's merge the rest to catch the lock if it happens again ( all the `"$(` stuff, while annoying was a red herring )